### PR TITLE
Allow object serialization on DefaultSdkAutoConstructList and Default…

### PR DESCRIPTION
**Pull request by udaysagar2177**
## Description
Allow Object serialization on DefaultSdkAutoConstructList and DefaultSdkAutoConstructMap.

## Motivation and Context
software.amazon.awssdk.services.ec2.model.InstanceStatus isn't serializable because DefaultSdkAutoConstructList instance doesn't implement Serializable.

## Testing
Wrote simple tests to check if DefaultSdkAutoConstructList & DefaultSdkAutoConstructMap are serializable and deserializable. They fail before the fix and pass after the fix.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](../docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by udaysagar2177** [comment]
@zoewangg Could you please take a look at this PR? Thanks!

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructList.java [hunk 1]
```diff
@@ -158,4 +162,8 @@ public int lastIndexOf(Object o) {
         return impl.subList(fromIndex, toIndex);
     }
 
+    // For deserialization
+    private Object readResolve() {
```
Comment (zoewangg) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructList.java]: Can you give us more context of why this is needed?

Comment (udaysagar2177) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructList.java]: This is a serialization trick to ensure that we use the singleton instance (INSTANCE on line 37) when we deserialize this class's object. (Usually, deserialization constructs a new object without readResolve())

Comment (udaysagar2177) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructList.java]: @zoewangg ping

Comment (zoewangg) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructList.java]: Apologies for the delay. Hmm, I'm not sure if we want to add this. This seems a bit hacky to me.

File: core/sdk-core/src/test/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructListTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package software.amazon.awssdk.core.util;
```
Comment (zoewangg) [inline core/sdk-core/src/test/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructListTest.java]: The test files are missing copyright header. You can copy paste the copyright header from other file in this repo

**Comment by dagnir** [comment]
Closing this PR for now as it seems it hasn't been update since @zoewangg made some comments. Please feel free to reopen once it's updated and ready for another look.

**Pull request closed without merging**